collect_app/src/main/java/org/odk/collect/android/projects/ProjectImporter.kt

```diff
@@ -1,33 +1,67 @@
 package org.odk.collect.android.projects
 
-import org.odk.collect.android.preferences.keys.MetaKeys
+import android.content.Context
+import androidx.preference.PreferenceManager
+import org.apache.commons.io.FileUtils.moveDirectoryToDirectory
+import org.odk.collect.android.preferences.source.SettingsProvider
+import org.odk.collect.android.storage.StoragePathProvider
+import org.odk.collect.android.utilities.FileUtils
 import org.odk.collect.projects.Project
+import org.odk.collect.projects.Project.Saved
 import org.odk.collect.projects.ProjectsRepository
-import org.odk.collect.shared.Settings
+import java.io.File
+import java.io.FileNotFoundException
 
 class ProjectImporter(
+    private val context: Context,
+    private val storagePathProvider: StoragePathProvider,
     private val projectsRepository: ProjectsRepository,
-    private val metaSettings: Settings
+    private val settingsProvider: SettingsProvider
 ) {
     fun importDemoProject() {
-        val project = Project("Demo project", "D", "#3e9fcc", DEMO_PROJECT_ID)
+        val project = Saved(DEMO_PROJECT_ID, "Demo project", "D", "#3e9fcc")
         projectsRepository.save(project)
-        metaSettings.save(MetaKeys.CURRENT_PROJECT_ID, DEMO_PROJECT_ID)
+        setupProject(project)
     }
 
-    // Now it does the same like importDemoProject() but it should be changed later
-    fun importExistingProject() {
-        val project = Project("Demo project", "D", "#3e9fcc", DEMO_PROJECT_ID)
-        projectsRepository.save(project)
-        metaSettings.save(MetaKeys.CURRENT_PROJECT_ID, DEMO_PROJECT_ID)
+    fun importExistingProject(): Saved {
```

Comment:
I feel uneasy with this solution a user can have a lot of files so this operation might be time-consuming. Shouldn't this be a service like we did in the case of migrating files between storages?

Reply:
This had crossed my mind as well - we might just always get away with it as it's a move rather than a copy but I do feel uncomfortable doing IO on the UI thread like this. It would be good for us to test this out with a large amount of files. How about I add an issue to optimize this with a "needs discussion" on it (with a note for us to test this)?

Reply:
Ok issue [here](https://github.com/getodk/collect/issues/4578). I realized when writing it that we might be able to just run `ApplicationInitializer` off the UI thread to make these kinds of things easier.